src/pages/SupplierInvoicePage.js

```diff
@@ -107,56 +208,234 @@ export class SupplierInvoicePage extends GenericPage {
           title: `${pageInfoStrings.confirm_date}:`,
           info: formatDate(transaction.confirmDate),
         },
+        {
+          title: `${pageInfoStrings.total_price}:`,
+          info: this.state.totalPrice,
+        },
       ],
       [
         {
+          title: `${pageInfoStrings.supplier}:`,
+          info: this.props.transaction.otherParty && this.props.transaction.otherParty.name,
+        },
+        {
           title: `${pageInfoStrings.their_ref}:`,
           info: transaction.theirRef,
+          onPress: this.openTheirRefEditor,
+          editableType: 'text',
         },
         {
           title: `${pageInfoStrings.comment}:`,
           info: transaction.comment,
+          onPress: this.openCommentEditor,
+          editableType: 'text',
         },
       ],
     ];
-    return <PageInfo columns={infoColumns} />;
+    return (
+      <PageInfo columns={infoColumns} isEditingDisabled={this.props.transaction.isFinalised} />
+    );
   }
 
-  renderCell(key, transactionItem) {
+  renderCell(key, transactionBatch) {
+    const isEditable = !this.props.transaction.isFinalised;
+    const type = isEditable ? 'editable' : 'text';
+    const editableCell = {
+      type: type,
```

Comment:
To minimise mistake (foul play), on receiving end -> I don't think batch name, batch expiry, pack size, cost price, should be editable (since normal supplier invoice-> inter-store transfer and stock transfer from central store should set this accurately, it's good to see this fields though). I can start an e-mail chain to see what everyone else thinks?